proposals/299-ip-failure-count.txt

```diff
@@ -29,44 +29,42 @@ Ticket: https://trac.torproject.org/projects/tor/ticket/27491
    in the random decision used for choosing an IP family and the data fields
    used by the algorithm.
 
-2. Failure Counter Design
+2. Options To Enable The Failure Counter
 
-   I propose that the failure counter uses the following fields:
+   To enable the failure counter, we will add a flags to ClientAutoIPv6ORPort.
+   The new format for ClientAutoIPv6ORPort is:
 
-      * IPv4 failure count
+      ClientAutoIPv6ORPort 0|1 [flags]
 
-      * IPv4 failure count from no route (autofail)
+   The first argument is to enable the automatic selection between IPv4 and
+   IPv6 if it is 1. The second argument is a list of optional flags.
 
-      * IPv6 failure count
+   The only flag so far is "TrackFailures", which enables the tracking of
+   failures to make a better decision when selecting between IPv4 and IPv6.
+   The tracking of failures will be described in the rest of this proposal.
 
-      * IPv6 failure count from no route (autofail)
+   However, we should be open to more flags from future proposals as they
+   are written and implemented.
 
-   These entries will exist as internal counters for the current session, and
-   as a list of the previous counts of these counters from previous sessions in
-   the state file.
+3. Failure Counter Design
 
-   These values will be stored as 32-bit unsigned integers for the current
-   session, and as comma seperated values in the statefile.
+   I propose that the failure counter uses the following fields:
+
+      * IPv4 failure points
 
-   The list capacity will be the 4 most recent sessions for each field above
-   stored in the state file with the least recent first. This is for the
-   following reasons:
+      * IPv6 failure points
 
-      * We can forget about the oldest sessions without having to keep the
-        exact timestamp when a client failed. This prevents an attacker from
-        getting detailed failure information from the state file.
+   These entries will exist as internal counters for the current session, and
+   a calculated value from the previous session in the statefile. 
 
-      * Some clients (mobile phones, laptops) may switch between networks of
-        which may be more or less reliable than the previous in terms of IPv4
-        or IPv6. In this case, it makes less sense to remember old failures
-        caused on a different network from the current one.
+   These values will be stored as 32-bit unsigned integers for the current
```

Comment:
If the values never go over 100, then we can store them as 16-bit unsigned integers.
But the difference really isn't that important.